File: 2p-net/trunk/examples/.NET from Prolog/Students/VBStudent/Student.vb
Namespace VBStudent
    Public Class Student
        Private _id As Integer

        Private _name As String

        Private _surname As String

        Private _exams As Integer

        Private Shared _staticProp As String = "Static Prop"

        Public publicField As String

        Public Shared publicStaticField As String = "Static Field"

        Public Shared Property StaticProperty() As String
            Get
                Return Student._staticProp
            End Get
            Set(value As String)
                Student._staticProp = value
            End Set
        End Property

        Public Property Id() As Integer
            Get
                Return Me._id
            End Get
            Set(value As Integer)
                Me._id = value
            End Set
        End Property

        Public Property Name() As String
            Get
                Return Me._name
            End Get
            Set(value As String)
                Me._name = value
            End Set
        End Property

        Public Property Surname() As String
            Get
                Return Me._surname
            End Get
            Set(value As String)
                Me._surname = value
            End Set
        End Property

        Public Property Exams() As Integer
            Get
                Return Me._exams
            End Get
            Set(value As Integer)
                Me._exams = value
            End Set
        End Property

        Public Sub New(id As Integer, name As String, surname As String)
            Me._exams = 10
            Me.publicField = "Public Field"
            Me.Id = id
            Me.Name = name
            Me.Surname = surname
        End Sub

        Public Function PrintStudent() As String
            Return Me.Id.ToString() + Me.Name + Me.Surname
        End Function

        Public Shared Function PrintInfoUni() As String
            Return "Alma Mater Studiorum - Bologna"
        End Function
    End Class
End Namespace

File: 2p-net/trunk/examples/.NET from Prolog/Students/CStudent/Student.cs
namespace CStudent
{

    public class Student
    {
        private string _Name;
        private string _SecondName;
        private int _Matricola;
        private int _exams = 10;
        public string publicField = "Public Filed";
        private static string _staticProp = "Static Property";
	    public static string publicStaticField = "Public Static Filed";

        public Student(int matricola, string name, string secondName)
        {
            _Name = name;
            _SecondName = secondName;
            _Matricola = matricola;
        }
        public string Name
        {
            get { return _Name; }
            set { _Name = value; }
        }

        public string Surname
        {
            get { return _SecondName; }
            set { _SecondName = value; }
        }

        public int Id
        {
            get { return _Matricola; }
            set { _Matricola = value; }
        }

        public int Exams
        {
            get
            {
                return this._exams;
            }
            set
            {
                this._exams = value;
            }
        }

        public static string StaticProperty
        {
            get
            {
                return Student._staticProp;
            }
            set
            {
                Student._staticProp = value;
            }
        }

        public override string ToString()
        {
            return Name + " " + Surname + " " + Id.ToString();
        }

        public static string PrintInfoUni()
        {
            return "Alma Mater Studiorum - Bologna";
        }

        public string PrintStudent()
        {
            return this.Id + this.Name + this.Surname;
        }

        public string SignExam(int grade)
        {
            if (grade == 30)
            {
                return "Good job!";
            }
            return "Keep studying!";
        }
    }

}